### ISSUE TITLE
Completion and adoption of ST-tabbyAPI-loader into ST codebase

### ISSUE THREAD

**Issue by Dakraid** [theroyallab/ST-tabbyAPI-loader#8]
Greetings,
so I've recently started an effort to integrate model swapping into SillyTavern directly before I was made aware of this extension. Given I've only added the model swapping at this point, it isn't as feature-complete as this extension.

To avoid doing duplicate work, I would like to propose that I finish the extension while adding its code to the main ST code. Would this be appreciated, and if so are there certain wishes or points that I should respect?

With kind regards,
Kris

**Comment by kingbri1** [comment]
Yes, the point of the extension is to eventually integrate it into SillyTavern. I've been in discussion with @Cohee1207 and would like to get the ball rolling.

The extension is pretty much complete apart from lora selection and scaling parameters. So, I think it's ok to port over to ST staging in its current state. Can you please forward the work you've done and we can get started!

Also, feel free to ping me `@kingbri` on discord in ST's server or tabby's support server. It should be easier to discuss things there.

**Comment by Dakraid** [comment]
Sorry for the late response, I wasn't feeling well. Most of the work so far is here: https://github.com/Dakraid/SillyTavern/tree/feature/tabbyapi-switch-models

Cohee has already given me some pointers on how to improve things, and once that is done I'm going to add the admin key check and adopt the rest of the extension.
